controller/up.go

```diff
@@ -7,38 +7,100 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"io/fs"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/railwayapp/cli/entity"
 	gitignore "github.com/railwayapp/cli/gateway"
 )
 
+var validIgnoreFile = map[string]bool{
+	".gitignore":     true,
+	".railwayignore": true,
+}
+
+var skipDirs = []string{
+	".git",
+	"node_modules",
+}
+
+type ignoreFile struct {
+	prefix string
+	ignore *gitignore.GitIgnore
+}
+
+func scanIgnoreFiles(src string) ([]ignoreFile, error) {
+	ret := []ignoreFile{}
+
+	if err := filepath.WalkDir(src, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			// no sense scanning for ignore files in skipped dirs
+			for _, s := range skipDirs {
+				if filepath.Base(path) == s {
+					return filepath.SkipDir
+				}
+			}
+
+			return nil
+		}
+
+		fname := filepath.Base(path)
+		if validIgnoreFile[fname] {
+			igf, err := gitignore.CompileIgnoreFile(path)
+			if err != nil {
+				return err
+			}
+
+			prefix := filepath.Dir(path)
+			if prefix == "." {
+				prefix = "" // Handle root dir properly.
+			}
+
+			ret = append(ret, ignoreFile{
+				prefix: prefix,
+				ignore: igf,
+			})
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return ret, nil
+}
+
 func compress(src string, buf io.Writer) error {
 	// tar > gzip > buf
 	zr := gzip.NewWriter(buf)
 	tw := tar.NewWriter(zr)
-	ignore, err := gitignore.CompileIgnoreFile(".gitignore")
 
+	// find all ignore files, including those in subdirs
+	ignoreFiles, err := scanIgnoreFiles(src)
```

Comment:
ignoredFilenames, same as the return type

Reply:
See above.